### PR TITLE
Main Menu Page History Fix

### LINKED ISSUE #2593: Slimefun guide history doesn't retain pagination for the main menu
## :round_pushpin: Description (REQUIRED)

Hi, I have issue (reported by players) in SF guide. When players open category on another page, for ex. on page 2, then when they click back button they aren't returned to previous page (the second), but to the first one, always. And they probably should be returned back to previous page, so page 2, it's a little bit annoying for them when they open category on page 2, then they click on back, appear on the first page, and have to click again on page 2 and then on another category... and imagine you're searching through categories and this really annoys you.

Same for sf cheat.

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

1. Give yourselves sf guide.
2. Open it.
3. Go on the next page (for ex. 2, may you need to add addons there).
4. Click on previous page (back button).
5. Now you're on the first page instead on the second one.
https://streamable.com/cmixva

## :bulb: Expected behavior (REQUIRED)

It should go back to the second page, not to the first one, because it's annoying for players when they have to click and click through pages and then categories again and again.

## :compass: Environment (REQUIRED)

 - Server Software (Spigot/Paper): Paper
 - Minecraft Version: 1.16.4
 - Slimefun Version: 765
 - CS-CoreLib Version: 92

### PR DESCRIPTION
**Pull request by Mooy1**
## Description
Keeps the main menu page history saved

## Proposed changes
Added a main menu page field to GuideHistory, added getters and setters. When the guide is opened, it sets the main menu page to the page it was opened to.

## Related Issues (if applicable)
Fixes #2593

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java [hunk 1]
```diff
@@ -146,6 +146,7 @@ public void openMainMenu(PlayerProfile profile, int page) {
         if (isSurvivalMode()) {
             profile.getGuideHistory().clear();
         }
+        profile.getGuideHistory().setMainMenuPage(page);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java:149]: The guide history should not be touched in non-survival mode.
Can also store the guide history as a variable here if we're being honest.
```suggestion
        if (isSurvivalMode()) {
            profile.getGuideHistory().clear();
            profile.getGuideHistory().setMainMenuPage(page);
        }
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Nice little fix actually, thanks!
Just one remark I'd like to leave.

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/guide/GuideHistory.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import java.util.Deque;
 import java.util.LinkedList;
+import java.util.Objects;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/guide/GuideHistory.java:5]: Unused import

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java [hunk 3]
```diff
@@ -145,6 +145,7 @@ public void openMainMenu(PlayerProfile profile, int page) {
 
         if (isSurvivalMode()) {
             profile.getGuideHistory().clear();
+            profile.getGuideHistory().setMainMenuPage(page);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java:148]: We could store `GuideHistory` as a variable here.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

**Review by svr333 — APPROVED** [review]
LGTM

**Pull request merged**